src/OrbitGl/TranslationStack.h

```diff
@@ -8,22 +8,30 @@
 #include <vector>
 
 #include "CoreMath.h"
+#include "Geometry.h"
 
 namespace orbit_gl {
+
+struct Vec2Z {
+  Vec2 shape;
+  float z{};
+};
```

Comment:
I minimize the use of Vec3 because if also allows for
1. Arithmetics that _surprisingly_ operates on the z-component. If you wish to do that -- walk extra mile, so everyone is sure you've wanted this. 
2. `float z = vertex[3]` // UB

The only thing I don't like about `Vec2Z` as it might read as "vector-to-Z", thus suggesting it's a map. Suggestions for better naming are welcome

Reply:
... `LayeredVec2` ?

Reply:
I like LayeredVec2 more. I was about to suggest Vec3 as well, but I get the reasoning behind this.

Reply:
I'm not sure either about this structure, I find it a bit confusing. Maybe replacing "shape" by "xy" would help, but to me "TranslationStack" should be as generic as possible, i.e. operate on Vec3s, and the flooring operation should be more explicitly named, like "TranslateAndFloorVertex" -> "TranslateAndFloorXY". In its current form, I find it strange that "TranslateAndFloorVertex" floors 2 out of 3 components.

Reply:
> Maybe replacing "shape" by "xy" would help

Good point. 

> flooring operation should be more explicitly named, like "TranslateAndFloorVertex" -> "TranslateAndFloorXY"

If we want to rename it like that, it should rather be `TranslateXYZAndFloorXY`, as otherwise it's not clear if anything happens to `z` at all. @pierricgimmig , are you OK with that name? 

And another reason why I don't like using Vec3 here. Why is `z` float? It should be of ordered type, no doubt. But why not `uint`? @karupayun , WDYT? If we make `z` of `uint` type, it will be clear why we don't round it.

Reply:
Yeah that's a better name, works with me!

Reply:
z is float for historical reasons (because we draw in 3D). I'm ok with changing it but not in this PR for sure.